lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -370,40 +362,79 @@ void FrontendInputsAndOutputs::forEachInputProducingSupplementaryOutput(
     fn(firstInput());
 }
 
+bool FrontendInputsAndOutputs::hasSupplementaryOutputPath(
+    llvm::function_ref<const std::string &(const SupplementaryOutputPaths &)>
+        extractorFn) const {
+  bool hasPath = false;
+  forEachInputProducingSupplementaryOutput([&](const InputFile &input) -> void {
```

Comment:
Is there a way to make this early-exit? Maybe a variant of `forEachInputProducingSupplementaryOutput` that takes a bool-returning function_ref?

Reply:
Sure! Made a bunch of these use that convention. Mostly helps.